sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/CommunicationIdentityAsyncManagedIdentityTests.java

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.identity;
+
+import com.azure.communication.common.CommunicationUserIdentifier;
+import com.azure.communication.identity.models.CommunicationTokenScope;
+import com.azure.communication.identity.models.GetTokenForTeamsUserOptions;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.http.rest.Response;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static com.azure.communication.identity.CteTestHelper.skipExchangeAadTeamsTokenTest;
+
+public class CommunicationIdentityAsyncManagedIdentityTests extends CommunicationIdentityClientTestBase {
+
+    private CommunicationIdentityAsyncClient asyncClient;
+
+    @Test
+    public void createAsyncIdentityClient() {
+        // Arrange
+        CommunicationIdentityClientBuilder builder = createClientBuilderUsingManagedIdentity(httpClient);
+        asyncClient = setupAsyncClient(builder, "createAsyncIdentityClientUsingManagedIdentity");
+        assertNotNull(asyncClient);
+
+        // Action & Assert
+        Mono<CommunicationUserIdentifier> response = asyncClient.createUser();
+        StepVerifier.create(response)
+                .assertNext(item -> {
+                    assertNotNull(item.getId());
+                    assertFalse(item.getId().isEmpty());
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    public void createUserWithResponse() {
+        // Arrange
+        CommunicationIdentityClientBuilder builder = createClientBuilderUsingManagedIdentity(httpClient);
+        asyncClient = setupAsyncClient(builder, "createUserWithResponseUsingManagedIdentity");
+
+        // Action & Assert
+        Mono<Response<CommunicationUserIdentifier>> response = asyncClient.createUserWithResponse();
+        StepVerifier.create(response)
+                .assertNext(item -> {
+                    assertNotNull(item.getValue().getId());
+                    assertFalse(item.getValue().getId().isEmpty());
+                    assertEquals(201, item.getStatusCode(), "Expect status code to be 201");
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    public void createUserWithContext() {
```

Comment:
What is the difference between this test and `createAsyncIdentityClient`? I don't see any.